nondex-common/src/main/java/edu/illinois/nondex/shuffling/ControlNondeterminism.java

```diff
@@ -108,10 +108,7 @@ public void run() {
             } catch (IOException ioe) {
                 Logger.getGlobal().log(Level.SEVERE,
                         "IOException when printing shuffling counts in shutdown hook.", ioe);
-            } catch (Throwable ex) {
```

Comment:
Do not remove this code. This catch clause will catch and log errors at runtime, e.g., NullPointerException and other RuntimeExceptions that need not be declared and checked and log it properly.